### PR TITLE
ICKY-490-View and Select Previously Saved Shipping Details

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

File: components/KiboOrderSummary.vue [hunk 1]
```diff
@@ -1,106 +1,68 @@
...
-          total,
-          shippingMethod,
-          propertiesNames,
-        }"
-      >
+    <div class="sf-order-summary__heading">{{ orderTitle }}</div>
+    <div class="sf-order-summary__content">
+      <div class="props">
         <SfProperty
           :name="propertiesNames[0]"
```
Comment (kibo-chandradeeptalaha) [inline components/KiboOrderSummary.vue]: Do we need to make the name dynamic when the values are static ?

File: components/KiboOrderSummary.vue [hunk 2]
```diff
@@ -110,98 +72,112 @@ export default {
       type: Object,
       default: () => ({}),
     },
-    propertiesNames: {
```
Comment (kibo-chandradeeptalaha) [inline components/KiboOrderSummary.vue:113]: We are using this prop in template right ?

File: components/KiboOrderSummary.vue [hunk 3]
```diff
@@ -110,98 +72,112 @@ export default {
...
   },
-  setup(props) {
+  setup(props, context) {
     const { order } = props
 
-    const totalItems = computed(() => checkoutGetters.getLineItemTotal(order))
-    const shippingMethod = computed(() => checkoutGetters.getShippingMethod(order))
-    const subtotal = computed(() => checkoutGetters.getSubtotal(order))
-    const total = computed(() => checkoutGetters.getTotal(order))
+    const numberOfItems = computed(() => checkoutGetters.getLineItemTotal(order) || 0)
```
Comment (kibo-chandradeeptalaha) [inline components/KiboOrderSummary.vue]: can we move the || inside getter itself ?

**Review by kibo-chandradeeptalaha — CHANGES_REQUESTED** [review]
Please take a look

File: components/KiboOrderSummary.vue (hunk 2 above)
Comment (kibo-sushant) [reply components/KiboOrderSummary.vue:113]: No, we are not using it anymore (at least right now)

File: components/KiboOrderSummary.vue (hunk 3 above)
Comment (kibo-sushant) [reply components/KiboOrderSummary.vue]: Resolved

File: components/KiboOrderSummary.vue (hunk 1 above)
Comment (kibo-sushant) [reply components/KiboOrderSummary.vue]: Just have followed SfOrderSummary patten, but yes we can use it directly as well

File: components/UserSavedAddress.vue [hunk 4]
```diff
@@ -34,23 +34,24 @@
...
-          @change="$emit('onSelect', address)"
+          class="sf-radio"
+          selected=""
+          @input="onSelect"
         >
           <template #label>
             <div class="radio-button">
               <div v-if="isDefaultAddress()" class="is-primary">{{ $t("Primary") }}</div>
-              <p>{{ firstName }} {{ lastNameOrSurname }}</p>
+              <p>{{ firstName }} {{ lastName }}</p>
```
Comment (kibo-kevinwatts) [inline components/UserSavedAddress.vue]: shouldn't these stay, "postalOrZipCode", "lastNameOrSurname" "cityOrTown", etc

File: components/UserSavedAddresses.vue [hunk 5]
```diff
@@ -107,29 +118,29 @@ export default defineComponent({
       showAddressForm.value = true
     }
 
-    const onSelectedAdress = (address) => {
+    const selectAddress = async (address) => {
       activeAddress.value = address
-      context.emit("onSelect", activeAddress)
+      await context.emit("onSave", { ...activeAddress.value })
```
Comment (kibo-kevinwatts) [inline components/UserSavedAddresses.vue]: this emitter doesn't need to be async/awaited

File: components/UserSavedAddresses.vue [hunk 6]
```diff
@@ -107,29 +118,29 @@ export default defineComponent({
...
-      activeAddress.value = {}
     }
 
     const setInputAddressData = (address) => {
       activeAddress.value = { ...address }
     }
 
     const saveAddress = async () => {
-      await context.emit("onSave", activeAddress)
+      await context.emit("onSave", { ...activeAddress.value })
```
Comment (kibo-kevinwatts) [inline components/UserSavedAddresses.vue]: no need to be awaited

File: composables/useUserAddresses.ts [hunk 7]
```diff
@@ -5,8 +5,12 @@ import { useState, useNuxtApp } from "#app"
 export const useUserAddresses = () => {
   const nuxt = useNuxtApp()
   const fetcher = nuxt.nuxt2Context.$gqlFetch
-  const addresses = useState<Maybe<Array<CustomerContactCollection>>>(
-    `use-user-address-result`,
+  const userShippingAddresses = useState<Maybe<Array<CustomerContactCollection>>>(
```
Comment (kibo-kevinwatts) [inline composables/useUserAddresses.ts]: The types for both of the state references should be Maybe<Arrary<CustomerContact>>

File: lang/en.js [hunk 8]
```diff
@@ -182,7 +182,7 @@ export default {
   "Save payment method": "Save payment method",
   "Save Shipping Address": "Save shipping address",
   Info: "info",
-  "Shiping To Home": "Ship to Home",
+  "Shiping To Home": "Shiping to Home",
```
Comment (kibo-kevinwatts) [inline lang/en.js]: fix typo, Shipping

File: pages/checkout.vue [hunk 9]
```diff
@@ -117,7 +120,8 @@ import {
 } from "@/composables"
 import { useNuxtApp } from "#app"
 import { buildPaymentMethodInput, defaultPaymentDetails } from "@/composables/helpers"
-import { shopperContactGetters, shippingMethodGetters, checkoutGetters } from "~~/lib/getters"
+import { shopperContactGetters, shippingMethodGetters, checkoutGetters } from "@/lib/getters"
+import StoreLocatorModal from "@/components/StoreLocatorModal.vue"
```
Comment (kibo-kevinwatts) [inline pages/checkout.vue:132]: nuxt should be importing this modules for us automatically

File: pages/checkout.vue [hunk 10]
```diff
@@ -134,34 +138,34 @@ export default {
     const { locale } = context.root.$i18n
     const currencyCode = context.root.$i18n.getNumberFormat(locale)?.currency?.currency
 
+    const keyRefKiboOrderSummery = ref("initialKey")
```
Comment (kibo-kevinwatts) [inline pages/checkout.vue]: Why is this needed?

File: components/KiboOrderSummary.vue (hunk 1 above)
Comment (kibo-kevinwatts) [reply components/KiboOrderSummary.vue]: would it make more sense for this to be an object for labels? instead of an array?

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/KiboOrderSummary.vue (hunk 1 above)
Comment (kibo-sushant) [reply components/KiboOrderSummary.vue]: Resolved. Used Object instead of array

File: components/UserSavedAddress.vue (hunk 4 above)
Comment (kibo-sushant) [reply components/UserSavedAddress.vue]: Resolved.

File: components/UserSavedAddresses.vue (hunk 5 above)
Comment (kibo-sushant) [reply components/UserSavedAddresses.vue]: Resolved.

File: components/UserSavedAddresses.vue (hunk 6 above)
Comment (kibo-sushant) [reply components/UserSavedAddresses.vue]: Resolved.

File: composables/useUserAddresses.ts (hunk 7 above)
Comment (kibo-sushant) [reply composables/useUserAddresses.ts]: Resolved.

File: lang/en.js (hunk 8 above)
Comment (kibo-sushant) [reply lang/en.js]: Resolved.

File: pages/checkout.vue (hunk 10 above)
Comment (kibo-sushant) [reply pages/checkout.vue]: Resolved

File: pages/checkout.vue (hunk 9 above)
Comment (kibo-sushant) [reply pages/checkout.vue:132]: nuxt only auto importing components ( and not getters and  .vue files), hence imports are required here.

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]
